skimage/feature/blob.py

```diff
@@ -24,50 +23,58 @@ def _blob_overlap(blob1, blob2):
 
     Parameters
     ----------
-    blob1 : sequence
-        A sequence of ``(y,x,sigma)``, where ``x,y`` are coordinates of blob
-        and sigma is the standard deviation of the Gaussian kernel which
-        detected the blob.
-    blob2 : sequence
-        A sequence of ``(y,x,sigma)``, where ``x,y`` are coordinates of blob
-        and sigma is the standard deviation of the Gaussian kernel which
-        detected the blob.
+    blob1 : sequence of arrays
+        A sequence of ``(row, col, sigma)`` or ``(pln, row, col, sigma)``,
+        where ``row, col`` (or ``(pln, row, col, sigma)``) are coordinates
+        of blob and ``sigma`` is the standard deviation of the Gaussian kernel
+        which detected the blob.
+    blob2 : sequence of arrays
+        A sequence of ``(row, col, sigma)`` or ``(pln, row, col, sigma)``,
+        where ``row, col`` (or ``(pln, row, col, sigma)``) are coordinates
+        of blob and ``sigma`` is the standard deviation of the Gaussian kernel
+        which detected the blob.
 
     Returns
     -------
     f : float
-        Fraction of overlapped area.
+        Fraction of overlapped area (or volume in 3D).
     """
-    root2 = sqrt(2)
+    n_dim = len(blob1) - 1
+    root_ndim = sqrt(n_dim)
 
     # extent of the blob is given by sqrt(2)*scale
-    r1 = blob1[2] * root2
-    r2 = blob2[2] * root2
-
-    d = hypot(blob1[0] - blob2[0], blob1[1] - blob2[1])
+    r1 = blob1[-1] * root_ndim
+    r2 = blob2[-1] * root_ndim
 
+    d = sqrt(np.sum((blob1[:-1] - blob2[:-1])**2))
     if d > r1 + r2:
         return 0
 
     # one blob is inside the other, the smaller blob must die
     if d <= abs(r1 - r2):
         return 1
 
-    ratio1 = (d ** 2 + r1 ** 2 - r2 ** 2) / (2 * d * r1)
-    ratio1 = np.clip(ratio1, -1, 1)
-    acos1 = arccos(ratio1)
+    if n_dim == 2:
+        ratio1 = (d ** 2 + r1 ** 2 - r2 ** 2) / (2. * d * r1)
+        ratio1 = np.clip(ratio1, -1, 1)
+        acos1 = math.acos(ratio1)
 
-    ratio2 = (d ** 2 + r2 ** 2 - r1 ** 2) / (2 * d * r2)
-    ratio2 = np.clip(ratio2, -1, 1)
-    acos2 = arccos(ratio2)
+        ratio2 = (d ** 2 + r2 ** 2 - r1 ** 2) / (2. * d * r2)
```

Comment:
@emmanuelle why is the `.` necessary? If it's for Python 2, I would prefer a `from __future__ import division`